runtime/connectors/gcs/gcs.go

```diff
@@ -191,6 +200,100 @@ func (c connector) HasAnonymousAccess(ctx context.Context, env *connectors.Env,
 	return bucketObj.IsAccessible(ctx)
 }
 
+func (c Connector) ListBuckets(ctx context.Context, req *runtimev1.GCSListBucketsRequest, env *connectors.Env) ([]string, string, error) {
+	credentials, err := resolvedCredentials(ctx, env)
+	if err != nil {
+		return nil, "", err
+	}
+
+	client, err := storage.NewClient(ctx, option.WithCredentials(credentials))
+	if err != nil {
+		return nil, "", err
+	}
+	defer client.Close()
+
+	projectID := credentials.ProjectID
+	if projectID == "" {
+		f := &credentialsFile{}
+		if err := json.Unmarshal(credentials.JSON, f); err != nil {
+			return nil, "", err
+		}
+
+		projectID = f.getProjectID()
+	}
```

Comment:
Is `credentials.JSON` guaranteed to be set in this case?

Reply:
Yes. I took a look at the credentials resolution. There seem to be four cases where credentials can be retrieved from the host system. 
1. Via env variable : Json will be set
2. via well known file (application_default_credentials.json) : Json will be set
3. We're on a Google App Engine : project ID will be set, json will not be
4. We're on Google Compute Engine : project ID will be set, json will not be

Reply:
Sounds good. Just to be defensive for future changes, let's add a check for `credentials.JSON == nil` also, and return a nicer response for that case